sunscreen_fhe_program/src/lib.rs

```diff
@@ -383,9 +386,23 @@ impl FheProgramTrait for FheProgram {
 
         Self {
             data: self.data,
-            graph: CompilationResult(StableGraph::from(pruned)),
+
+            #[cfg(not(feature = "debugger"))]
+            graph: CompilationResult {
+                graph: StableGraph::from(pruned),
+            },
+
+            #[cfg(feature = "debugger")]
+            graph: CompilationResult {
+                graph: StableGraph::from(pruned),
+                metadata: DebugData::new(),
+            },
+
```

Comment:
I don't think a compilation result should have the counter or group stack. The context should have those.